scripts/ctest/ctest_driver.py

```diff
@@ -122,18 +169,22 @@ def run_single_test_suite(suite, ctest_path, cmake_build_path, build_config, dis
                 rp.rename_test_results(cmake_build_path, test_result_prefix, iteration + 1, repeat)
             if result.returncode:
                 error_code = result.returncode
+            if soft_repeat_timeout and (time.time() - start_time) > soft_repeat_timeout and (iteration + 1 != repeat):
+                print(f"Soft timeout of {soft_repeat_timeout} seconds exceeded, executed {iteration + 1}/{repeat} runs.")
+                total_runs = iteration + 1
```

Comment:
Total should increment outside of these if-checks

Reply:
This doesn't increment, it only sets the total_runs if the timeout is exceeded.